client/App.tsx

```diff
@@ -3,38 +3,67 @@ import { Track } from "../shared/classes";
 import Table from "./components/table/Table";
 
 const App = (): ReactElement => {
-
   const [tracks, setTracks] = useState<Track[]>();
   const [genres, setGenres] = useState<string[]>();
 
   useEffect(() => {
-    fetch('http://localhost:3000/tracks')
+    fetch("http://localhost:3000/tracks")
       .then((res) => res.json())
       .then((json) => setTracks(json))
       .catch((e) => console.log(e));
 
-      //fetch genres and setGenres here
+    // fetch genres and setGenres here
+    fetch("http://localhost:3000/genres")
+      .then((res) => res.json())
+      .then((json) => setGenres(json))
+      .catch((e) => console.log(e));
   }, []);
 
   const handleSort = (column: string, direction: string): void => {
     fetch(`http://localhost:3000/sort?column=${column}&direction=${direction}`)
-    .then((res) => res.json())
-    .then((json) => setTracks(json))
-    .catch((e) => console.log(e));
+      .then((res) => res.json())
+      .then((json) => setTracks(json))
+      .catch((e) => console.log(e));
   };
 
   const handleTestClick = function (): void {
-    console.log(tracks);
+    console.table(tracks);
+    console.log(genres);
+    if (genres) {
+      createGenreButtons(genres);
+    } else {
+      console.error("'genre' is null");
+    }
   };
 
+//create a button for each item in the array
+  function createGenreButtons(btnNames: string[]): void {
+    const buttonContainer = document.getElementById("button-container");
+
+    if (!buttonContainer) {
+      console.error("button-container div is null");
+      return;
+    }
+
+    btnNames.forEach((btnName) => {
+      const button = document.createElement("button");
+      button.textContent = btnName;
+      //show genre of clicked button in the console
+      button.addEventListener("click", () => {
+        console.log(btnName);
+      });
+      buttonContainer.appendChild(button);
+    });
+  }
+
   return (
     <>
       <h1>DJ Database</h1>
       <button onClick={handleTestClick}>TEST</button>
+      <div id="button-container"></div>
```

Comment:
Instead of using vanilla JS here, try looking into how array.map is used in React to create a series of similar components from an array. Something like {genres.map((genre) => (<button onClick={()=> console.log(genre)>{genre}</button>)}
https://www.freecodecamp.org/news/how-to-render-lists-in-react/

Reply:
To get this to work you will likely have to use a "logical short circuit" which there is an example of below. Essentially the example below says "Are there any 'tracks'? If not, then don't do anything. If there are, then display the Table".
https://dev.to/harlessmark/short-circuit-evaluation-with-react-3dn4